orchagent/routeorch.cpp

```diff
@@ -1672,52 +1714,63 @@ bool RouteOrch::addRoutePost(const RouteBulkContext& ctx, const NextHopGroupKey
         return false;
     }
 
-    /* next_hop_id indicates the next hop id or next hop group id of this route */
-    sai_object_id_t next_hop_id;
-
     if (m_fgNhgOrch->isRouteFineGrained(vrf_id, ipPrefix, nextHops))
     {
         /* Route is pointing to Fine Grained ECMP nexthop group */
         isFineGrained = true;
     }
-    else if (nextHops.getSize() == 0)
-    {
-        /* The route is pointing to a blackhole */
-        blackhole = true;
-    }
-    else if (nextHops.getSize() == 1)
+    /* Check that the next hop group is not owned by NhgOrch. */
+    else if (ctx.nhg_index.empty())
```

Comment:
As mentioned above, please have as the second else case and retain the existing indentation

Reply:
Done